src/Polly.Core/Simmy/Outcomes/OutcomeChaosPipelineBuilderExtensions.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System.Diagnostics.CodeAnalysis;
+
+namespace Polly.Simmy.Outcomes;
+
+/// <summary>
+/// Extension methods for adding outcome to a <see cref="ResiliencePipelineBuilder"/>.
+/// </summary>
+public static partial class OutcomeChaosPipelineBuilderExtensions
+{
+    /// <summary>
+    /// Adds a fault chaos strategy to the builder.
+    /// </summary>
+    /// <param name="builder">The builder instance.</param>
+    /// <param name="enabled">A value that indicates whether or not the chaos strategy is enabled for a given execution.</param>
+    /// <param name="injectionRate">The injection rate for a given execution, which the value should be between [0, 1].</param>
+    /// <param name="fault">The exception to inject.</param>
+    /// <returns>The builder instance with the retry strategy added.</returns>
+    public static ResiliencePipelineBuilder AddChaosFault(this ResiliencePipelineBuilder builder, bool enabled, double injectionRate, Exception fault)
+    {
+        Guard.NotNull(builder);
+
+        builder.AddFaultCore(new OutcomeStrategyOptions<Exception>
+        {
+            Enabled = enabled,
+            InjectionRate = injectionRate,
+            Outcome = new(fault)
+        });
+        return builder;
+    }
+
+    /// <summary>
+    /// Adds a fault chaos strategy to the builder.
+    /// </summary>
+    /// <param name="builder">The builder instance.</param>
+    /// <param name="enabled">A value that indicates whether or not the chaos strategy is enabled for a given execution.</param>
+    /// <param name="injectionRate">The injection rate for a given execution, which the value should be between [0, 1].</param>
+    /// <param name="faultGenerator">The exception generator delegate.</param>
+    /// <returns>The builder instance with the retry strategy added.</returns>
+    public static ResiliencePipelineBuilder AddChaosFault(
+        this ResiliencePipelineBuilder builder, bool enabled, double injectionRate, Func<ValueTask<Outcome<Exception>?>> faultGenerator)
```

Comment:
```suggestion
        this ResiliencePipelineBuilder builder, bool enabled, double injectionRate, Func<Exception?>> faultGenerator)
```

Can we simplify this? Or just keep the one above.

Reply:
That's a great idea